vtex-apps/store
# Fix ProfileChallenge logic

#### What is the purpose of this pull request?

Refactor ProfileChallenge to use the runtime session and check only if session's `isAuthenticated` property is `true`.

#### What problem is this solving?

Fix issues @salesfelipe was having with it when it was a sales assistant.

#### How should this be manually tested?

Scenario 1: Not logged in user

1. Open:  https://breno--storecomponents.myvtex.com/account
2. You should be redirected to /login

Scenario 2: Logged in user
1. Open:  https://breno--storecomponents.myvtex.com/
2. Login with an user
3. Open https://breno--storecomponents.myvtex.com/account
4. It should stay on the My Account page

Scenario 3: Sales assistant
1. Open with a Sales Assistant admin account https://breno--storecomponents.myvtex.com/
2. Login as a user
3. Open https://breno--storecomponents.myvtex.com/account
4. It should stay on the My Account page

#### Screenshots or example usage
n/a

#### Types of changes
